src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php

```diff
@@ -672,19 +672,13 @@ public function createOrFirst(array $attributes = [], array $values = [], array
      */
     public function updateOrCreate(array $attributes, array $values = [], array $joining = [], $touch = true)
     {
-        if (is_null($instance = (clone $this)->where($attributes)->first())) {
-            if (is_null($instance = $this->related->where($attributes)->first())) {
-                return $this->create(array_merge($attributes, $values), $joining, $touch);
-            } else {
-                $this->attach($instance, $joining, $touch);
-            }
-        }
-
-        $instance->fill($values);
+        return tap($this->firstOrCreate($attributes, $values), function ($instance) use ($values) {
```

Comment:
But you're not even using the `$joining` attribute? We need to be really careful we aren't just breaking stuff here.

Reply:
Nice catch! I'll fix it